### PR TITLE
Message Count Rewards

### PR DESCRIPTION
**Pull request by devnote-dev**
Simple rewards checker for each count goal that is reached, starting at 50. Everything is commented and readily editable.

### CODE REVIEW

**Comment by DZ-TM** [comment]
Seems perfect but it should be put under the already made message count cmd (in the code, at the bottom) cus it uses .* regex already so two .* regex is a waste

File: Commands/MessageCount/countRewards.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+*/}}
+
+{{$rewards := dict 50 ROLE-ID 75 ROLE-ID 100 ROLE-ID 125 ROLE-ID 150 ROLE-ID}}
+{{/* The rewards count goes from 50 messages and upwards, you can change/add to this if you want */}}
+
+{{if ($db := dbGet .User.ID "msgCount")}}
+{{/* Checks if they have a message count */}}
+
+    {{range $k, $v := $rewards -}}
+    {{/* Ranges the KEY:VALUE pairs of the rewards */}}
```
Comment (jo3-l) [inline Commands/MessageCount/countRewards.go]:
```suggestion
    {{- /* Ranges the KEY:VALUE pairs of the rewards */}}
```
comment blocks need to be stripped too

File: Commands/MessageCount/countRewards.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        {{- if and (ge (toInt $db.Value) $k) (not (hasRoleID $v)) -}}
+        {{/* Checks if the user has a message count >= each KEY of the rewards AND that they do not already have that role reward */}}
```
Comment (jo3-l) [inline Commands/MessageCount/countRewards.go]:
```suggestion
        {{- /* Checks if the user has a message count >= each KEY of the rewards AND that they do not already have that role reward */}}
```
ditto

File: Commands/MessageCount/countRewards.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+            {{- addRoleID $v -}}
+            {{/* Adds the role(s) corresponding to the VALUE of the rewards if true */}}
```
Comment (jo3-l) [inline Commands/MessageCount/countRewards.go]:
```suggestion
            {{- /* Adds the role(s) corresponding to the VALUE of the rewards if true */}}
```

File: Commands/MessageCount/countRewards.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+			      {{- print "Congrats! You messaged **" $db.Value "** times and earned the <@&" $v "> role!\n" -}}
+			      {{/* Sends the congrats message */}}
```
Comment (jo3-l) [inline Commands/MessageCount/countRewards.go]:
```suggestion
			      {{- /* Sends the congrats message */}}
```

File: Commands/MessageCount/countRewards.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+        {{- end -}}
+
+    {{- end}}
```
Comment (jo3-l) [inline Commands/MessageCount/countRewards.go]:
```suggestion
    {{ end}}
```
theres no need for this

**Review by jo3-l — CHANGES_REQUESTED** [review]
smh forgetting about the poor comments

**Comment by soudblox** [comment]
damn looks lit, can i use it now xd

**Pull request closed without merging**